File: TP's/TP 1/Palindromo.java
import java.util.Scanner;

class Q1 {
	public static boolean theEnd(String str) {

		if (str.length() == 3 && str.charAt(0) == 'F' && str.charAt(1) == 'I' && str.charAt(2) == 'M') {
			return true;
		} else {
			return false;
		}
	}

	public static boolean ehpalin(String str) {
		int esq = 0;
		int dir = str.length() - 1;
		while (esq < dir) {
			if (str.charAt(esq) != str.charAt(dir)) {
				return false;
			}
			esq++;
			dir--;
		}
		return true;
	}

	public static void main(String[] args) {
		Scanner sc = new Scanner(System.in);
		String palavra;

		palavra = "";

		while (!theEnd(palavra)) {

			palavra = sc.nextLine();
			if (ehpalin(palavra)) {
				System.out.println("SIM");
			} else {
				System.out.println("NAO");
			}

		}
		sc.close();
	}
}

File: TP's/TP 1/Palindromo.c
#include <stdio.h>

int theEnd(char str[], int len) {
    return (len == 3 && str[0] == 'F' && str[1] == 'I' && str[2] == 'M');
}

int ehpalin(char str[], int len) {
    int esq = 0;
    int dir = len - 1;

    while (esq < dir) {
        if (str[esq] != str[dir]) {
            return 0; 
        }
        esq++;
        dir--;
    }
    return 1;
}

int main() {
    char palavra[100];
    int len;

    fgets(palavra, sizeof(palavra), stdin);

    len = 0;
    while (palavra[len] != '\0') {
        len++;
    }

    if (len > 0 && palavra[len - 1] == '\n') {
        palavra[len - 1] = '\0';
        len--;
    }

    while (!theEnd(palavra, len)) {

        if (ehpalin(palavra, len)) {
            printf("SIM\n");
        } else {
            printf("NAO\n");
        }

        fgets(palavra, sizeof(palavra), stdin);

        len = 0;
        while (palavra[len] != '\0') {
            len++;
        }

        if (len > 0 && palavra[len - 1] == '\n') {
            palavra[len - 1] = '\0';
            len--;
        }
    }

    return 0;
}
